
wanboot needs the provider archives

proper solution will be done via https://github.com/openssl/openssl/pull/21244

diff --git a/providers/build.info b/providers/build.info
index 3b211bbb2f..b4a1e007cf 100644
--- a/providers/build.info
+++ b/providers/build.info
@@ -40,7 +40,11 @@ $LIBCOMMON=libcommon.a
 $LIBFIPS=libfips.a
 $LIBLEGACY=liblegacy.a
 $LIBDEFAULT=libdefault.a
-LIBS{noinst}=$LIBDEFAULT $LIBCOMMON
+IF[{- $disabled{shared} -}]
+    LIBS=$LIBDEFAULT $LIBCOMMON
+ELSE
+    LIBS{noinst}=$LIBDEFAULT $LIBCOMMON
+ENDIF
 
 # Enough of our implementations include prov/ciphercommon.h (present in
 # providers/implementations/include), which includes crypto/*_platform.h
